[TELCO-*] 4.14 New features (#63785)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

[id="ocp-4-14-networking-nmstate-vrf"]
==== Support for VRF specification in NMState's NodeNetworkConfigurationPolicy resource (Technology Preview)

> With this update, you can associate a Virtual Routing and Forwarding (VRF) instance with a network interface by using a `NodeNetworkConfigurationPolicy` custom resource. By associating a VRF instance with a network interface, you can support traffic isolation, independent routing decisions, and the logical separation of network resources. This is available as a Technology Preview feature.

Reviewer: "This" in final sentence is ambiguous.

For more information, see xref:../networking/k8s_nmstate/k8s-nmstate-updating-node-network-config.adoc#virt-example-host-vrf_k8s_nmstate-updating-node-network-config[Example: Network interface with a VRF instance node network configuration policy].

[id="ocp-414-broadcom-bcm57504-support"]
==== Support for Broadcom BCM57504 is now GA

Support for the Broadcom BCM57504 network interface controller is now available for the SR-IOV Network Operator.

> For more information, see xref:../networking/hardware_networks/about-sriov.adoc#supported-devices_about-sriov

Reviewer: This link is broken. Add link text.

[Supported devices].
[id="ocp-4-14-creating-subinterface"]
==== MAC-VLAN, IP-VLAN, and VLAN subinterface creation for pods

> With this release, the ability to create a MAC-VLAN, IP-VLAN and VLAN subinterface based on a master interface in a container namespace is generally available. This feature allows you to create the master interfaces as part of the pod network configuration in a separate network attachment definition, and base the VLAN, MACVLAN or IPVLAN on this interface without requiring the knowledge of the network configuration of the node. For more information, see xref:../networking/multiple_networks/configuring-additional-network.html#nw-about-configuring-master-interface-container_configuring-additional-network[About configuring the master interface in the container network namespace].

Reviewer: `Master`'s fine here given context and existing use in published docs.

Reviewer: Consider more user-focused language, e.g.:

> With this release, you can now create MAC-VLAN, IP-VLAN, and VLAN subinterfaces based on master namespaces in container namespaces.

Additionally, make sure to use serial commas.

> With this release, the ability to create a MAC-VLAN, IP-VLAN, and [or?] VLAN subinterface based on a master interface in a container namespace is generally available.

[id="ocp-4-14-tap-device-plugin"]

> ==== Enhance network flexibility using the TAP device plugin

Author: plugin is ok

> This release introduces a new Container Network Interface (CNI) network plugin type: the tap device plugin. This plugin allows the creation of tap devices within containers. A tap device allows user space programs to handle network frames, acting as an interface that receives frames from and sends frames to user space applications instead of through traditional network interfaces. For more information, see xref:../networking/multiple_networks/configuring-additional-network.html#nw-multus-tap-object_configuring-additional-network[Configuration for a TAP additional network].

Author: I believe Red Hat changed guidance and plugin is accepted. I have checked the docs and plugin is used.

> This release introduces a new Container Network Interface (CNI) network plugin type: the tap device plugin. This plugin allows the creation of tap devices within containers. A tap device allows user-space programs to handle network frames, acting as an interface that receives frames from and sends frames to user-space applications instead of through traditional network interfaces. For more information, see xref:../networking/multiple_networks/configuring-additional-network.html#nw-multus-tap-object_configuring-additional-network[Configuration for a TAP additional network].

Reviewer: Is it `TAP` or `tap`? Heading and paragraph use difference capitalization schemes.

Author: Going to go with TAP strands for  "Terminal Access Point."

[id="ocp-4-14-non-root-dpdk"]
==== Support for running rootless DPDK workloads with kernel access using the TAP CNI plugin

> In {product-title} version 4.14 and later, DPDK applications that need to inject some traffic to the kernel can be run in non-privileged pods with the help of the tap CNI plugin. For more information, see xref:../networking/hardware_networks/using-dpdk-and-rdma.html#nw-running-dpdk-rootless-tap_using-dpdk-and-rdma[Using the TAP CNI to run a rootless DPDK workload with kernel access].

Author: plugin acceptable

> In {product-title} version 4.14 and later, DPDK applications that need to inject some traffic to the kernel can run in non-privileged pods with the help of the tap CNI plugin. For more information, see xref:../networking/hardware_networks/using-dpdk-and-rdma.html#nw-running-dpdk-rootless-tap_using-dpdk-and-rdma[Using the TAP CNI to run a rootless DPDK workload with kernel access].

Reviewer: Same issue with `TAP` consistency here.

You can enable `AdminNetworkPolicy` and `BaselineAdminNetworkPolicy` resources, which are part of the Network Policy V2 API, in clusters running the OVN-Kubernetes CNI plugin. Cluster administrators can apply cluster-scoped policies and safeguards for an entire cluster before namespaces are created. Network administrators can secure clusters by enforcing network traffic controls that cannot be overridden by users. Network administrators can enforce optional baseline network traffic controls that can be overridden by users in the cluster, if necessary. Currently, these APIs support only expressing policies for intra-cluster traffic.

[id="ocp-4-14-creating-subinterface"]
==== MAC-VLAN, IP-VLAN, and VLAN subinterface creation for pods

> With this release, the ability to create a MAC-VLAN, IP-VLAN, and VLAN subinterface based on a master interface in a container namespace is generally available. This feature lets you create master interfaces within the pod network configuration using a separate network attachment definition. You can then base the VLAN, MACVLAN or IPVLAN on this interface without knowing the network configuration of the node. For more information, see xref:../networking/multiple_networks/configuring-additional-network.html#nw-about-configuring-master-interface-container_configuring-additional-network[About configuring the master interface in the container network namespace].

Reviewer 2: ```suggestion
With this release, the ability to create a MAC-VLAN, IP-VLAN, and VLAN subinterface based on a master interface in a container namespace is generally available. This feature lets you create master interfaces within the pod network configuration using a separate network attachment definition. You can then base the VLAN, MACVLAN or IPVLAN on this interface without knowing the network configuration of the node. For more information, see xref:../networking/multiple_networks/configuring-additional-network.html#nw-about-configuring-master-interface-container_configuring-additional-network[About configuring the master interface in the container network namespace].
```
Still not a fan of the "This feature lets you create" construction, which goes explicitly against the ISG guidance against anthropomorphism: https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism. Is there a reason why this can't be revised so that "feature" isn't the main agent of action here and the user is?

Author: OK I resolved that thanks @bburt-rh

For further information about gathering debugging information relating to low-latency tuning, see xref:../scalability_and_performance/cnf-low-latency-tuning.adoc#cnf-collecting-low-latency-tuning-debugging-data-for-red-hat-support_cnf-master[Collecting low latency tuning debugging data for Red Hat Support].

[id="ocp-4-14-NRO-image-must-gather"]
==== Collecting data for the NUMA Resources Operator with the `must-gather` image of the operator

> In this release, the `must gather` tool has been updated to collect the NUMA-aware scheduling and NUMA Resources Operator data with the NUMA Resources Operator `must-gather` image.

Reviewer 3: "..to collect the NUMA-aware scheduling and NUMA Resources Operator data with .."
  I think it's better to have it as:
  ".. to collect the data of the NUMA Resources Operator with the `must-gather` image of the operator, formerly the `must-gather` image of performance-addon-operator was used for this purpose." 
  ^ I added the previous way of getting this data, not sure if that needs to be mentioned here, so leaving it up to you to decide whether or not to add it.

> For further information about gathering debugging information relating to the NUMA Resources Operator, see xref:../scalability_and_performance/cnf-numa-aware-scheduling.html#cnf-about-collecting-nro-data_numa-aware[Collecting NUMA-aware pod scheduling data].

Reviewer 3: same here concerning the term "pod scheduling", if we can replace it with NUMA Resources Operator data it would be good.

With this release, control plane machine sets are supported for Nutanix clusters.

For more information, see xref:../machine_management/control_plane_machine_management/cpmso-getting-started.adoc#cpmso-getting-started[Getting started with the Control Plane Machine Set Operator].
[id="ocp-4-14-mco-ca-distribution"]
==== Handling of registry certificate authorities

> The Machine Config Operator now handles distributing certificate authorities for image registries. This change does not affect end users.

Reviewer: Is it important to include this in the release notes, in that case?

Author: This is not my change so unable to comment

Author: This is the JIRA requesting that it is documented https://issues.redhat.com/browse/OSDOCS-8026

For further information about gathering debugging information relating to low-latency tuning, see xref:../scalability_and_performance/cnf-low-latency-tuning.adoc#cnf-collecting-low-latency-tuning-debugging-data-for-red-hat-support_cnf-master[Collecting low latency tuning debugging data for Red Hat Support].

[id="ocp-4-14-NRO-image-must-gather"]
==== Collecting data for the NUMA Resources Operator with the `must-gather` image of the operator

In this release, the `must gather` tool is updated to collect the data of the NUMA Resources Operator with the `must-gather` image of the operator.

> For further information about gathering debugging information relating to the NUMA Resources Operator, see xref:../scalability_and_performance/cnf-numa-aware-scheduling.html#cnf-about-collecting-nro-data_numa-aware[Collecting NUMA Resources Operator data].

Reviewer: Reduce ambiguity w/restrictive clause. Ex:

```suggestion
For further information about gathering debugging information that is related to the NUMA Resources Operator, see xref:../scalability_and_performance/cnf-numa-aware-scheduling.html#cnf-about-collecting-nro-data_numa-aware[Collecting NUMA Resources Operator data].
```

[id="ocp-4-14-additional-power-savings-control"]
==== Enabling more control over the c-states for each pod

> With this release, users are given more control over the c-states for their pods. Now instead of just disabling c-states completely users are able to specify a maximum latency in microseconds for the C-states. The additional configuration is available in the `cpu-c-states.crio.io` annotation and facilitates power savings in the high-priority applications by enabling some of the shallower c-states, instead of disabling them completely.

Reviewer: Watch comma use.

```suggestion
With this release, users are given more control over the c-states for their pods. Now, instead of disabling c-states completely, users are able to specify a maximum latency in microseconds for the C-states. The additional configuration is available in the `cpu-c-states.crio.io` annotation and facilitates power savings in the high-priority applications by enabling some of the shallower c-states, instead of disabling them completely.
```

Inconsistent capitalization of "c-states"/"C-states".

For further information about gathering debugging information relating to low-latency tuning, see xref:../scalability_and_performance/cnf-low-latency-tuning.adoc#cnf-collecting-low-latency-tuning-debugging-data-for-red-hat-support_cnf-master[Collecting low latency tuning debugging data for Red Hat Support].

[id="ocp-4-14-NRO-image-must-gather"]
==== Collecting data for the NUMA Resources Operator with the `must-gather` image of the operator

> In this release, the `must gather` tool is updated to collect the data of the NUMA Resources Operator with the `must-gather` image of the operator.

Reviewer 2: ```suggestion
In this release, the `must-gather` tool is updated to collect the data of the NUMA Resources Operator with the `must-gather` image of the operator.
```

For further information about gathering debugging information that is related to the NUMA Resources Operator, see xref:../scalability_and_performance/cnf-numa-aware-scheduling.html#cnf-about-collecting-nro-data_numa-aware[Collecting NUMA Resources Operator data].

[id="ocp-4-14-additional-power-savings-control"]
==== Enabling more control over the c-states for each pod

> With this release, users are given more control over the c-states for their pods. Now, instead of disabling c-states completely users are able to specify a maximum latency in microseconds for the c-states. The additional configuration is available in the `cpu-c-states.crio.io` annotation and facilitates power savings in the high-priority applications by enabling some of the shallower c-states, instead of disabling them completely.

Reviewer 2: ```suggestion
With this release, you have more control over the c-states for your pods. Now, instead of disabling c-states completely, you can specify a maximum latency in microseconds for c-states. You can configure this option in the `cpu-c-states.crio.io` annotation, which helps to optimize power savings in high-priority applications by enabling some of the shallower c-states instead of disabling them completely.
```
- Re-cast in second-person in a similar manner to the other list items. Revise to make more centered around user actions.
- Should "c-states" be in backticks?

Author: Looks like capitalized without backticks
